Userland/mount.cpp

```diff
@@ -0,0 +1,47 @@
+#include <LibCore/CArgsParser.h>
+#include <sys/stat.h>
+#include <stdio.h>
+#include <unistd.h>
+
+// This version of 'mount' must have the following arguments
+// 'mount <device_path> <mount_point>
+// It can currently only mount _physical_ devices found in /dev
+//
+// Currently, it is only possible to mount ext2 devices. Sorry! :^)
+int main(int argc, char** argv)
+{
+    CArgsParser args_parser("mount");
+    args_parser.add_arg("devname", "device path");
+    args_parser.add_arg("mountpoint", "mount point");
+    CArgsParserResult args = args_parser.parse(argc, argv);
+
+    if (argc == 3) {
+        // Let's use lstat so we can convert devname into a major/minor device pair!
+        struct stat st;
+        int rc = lstat(argv[1], &st);
+        if (rc < 0) {
+            perror("lstat");
+            return 1;
+
+        }
+
+        /**
+        // Make sure we're trying to mount a block or character device
```

Comment:
It's not valid to mount a character device :)
The defining trait of a character device is that it can only be read sequentially. Only block devices support random access, which is definitely necessary for mounting a file system.